templates/cluster-template-multiaz.yaml

```diff
@@ -0,0 +1,284 @@
+spec:
+  template:
+    spec:
+      node:
+        image:
+          name: "${OSC_IMAGE_NAME}"
+        vm:
+          rootDisk:
+            rootDiskSize: ${OSC_VOLUME_SIZE}
+            rootDiskIops: ${OSC_IOPS}
+            rootDiskType: "${OSC_VOLUME_TYPE}"
+          keypairName: "${OSC_KEYPAIR_NAME}"
+          subregionName: ${OSC_REGION}b
+          vmType: "${OSC_VM_TYPE}"
+---
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: MachineDeployment
+metadata:
+  name: "${CLUSTER_NAME}-md-c"
+spec:
+  clusterName: "${CLUSTER_NAME}"
+  replicas: ${WORKER_MACHINE_COUNT}
+  selector:
+    matchLabels:
+  template:
+    spec:
+      clusterName: "${CLUSTER_NAME}"
+      version: "${KUBERNETES_VERSION}"
+      bootstrap:
+        configRef:
+          name: "${CLUSTER_NAME}-md"
+          apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
+          kind: KubeadmConfigTemplate
+      infrastructureRef:
+        name: "${CLUSTER_NAME}-md-c"
+        apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+        kind: OscMachineTemplate
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: OscMachineTemplate
+metadata:
+  name: "${CLUSTER_NAME}-md-c"
+spec:
+  template:
+    spec:
+      node:
+        image:
+          name: "${OSC_IMAGE_NAME}"
+        vm:
+          rootDisk:
+            rootDiskSize: ${OSC_VOLUME_SIZE}
+            rootDiskIops: ${OSC_IOPS}
+            rootDiskType: "${OSC_VOLUME_TYPE}"
+          keypairName: "${OSC_KEYPAIR_NAME}"
+          subregionName: ${OSC_REGION}c
+          vmType: "${OSC_VM_TYPE}"
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: OscMachineTemplate
+metadata:
+  name: "${CLUSTER_NAME}-control-plane"
+spec:
+  template:
+    spec:
+      node:
+        image:
+          name: "${OSC_IMAGE_NAME}"
+        vm:
+          keypairName: "${OSC_KEYPAIR_NAME}"
+          rootDisk:
+            rootDiskSize: ${OSC_VOLUME_SIZE}
+            rootDiskIops: ${OSC_IOPS}
+            rootDiskType: "${OSC_VOLUME_TYPE}"
+          role: controlplane
+          vmType: "${OSC_VM_TYPE}"
+          tags:
+            osc.fcu.repulse_server: vm-control-plane
+---
+apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
+kind: KubeadmConfigTemplate
+metadata:
+  name: "${CLUSTER_NAME}-md"
+spec:
+  template:
+    spec:
+      files:
+      - content: |
+          #!/bin/sh
+
+          curl https://github.com/opencontainers/runc/releases/download/v1.1.1/runc.amd64 -Lo /tmp/runc.amd64
+          chmod +x /tmp/runc.amd64
+          cp -f /tmp/runc.amd64 /usr/local/sbin/runc
+        owner: root:root
+        path: /tmp/set_runc.sh
+        permissions: "0744"
+      joinConfiguration:
+        nodeRegistration:
+          name: "{{ ds.meta_data.local_hostname }}"
+          kubeletExtraArgs:
+            cloud-provider: external
+            provider-id: aws:///'{{ ds.meta_data.placement.availability_zone }}'/'{{ ds.meta_data.instance_id }}'
+      preKubeadmCommands:
+      - sh /tmp/set_runc.sh
```

Comment:
I think the runc part could be ommited (I'm testing this right now)

@pli01 you don't have it aither, correct?

Or, it should be in the packer build: https://github.com/kubernetes-sigs/image-builder

we would save 12 lines \o/

;)

Reply:
I'll keep the runc part for now. It is already present in the default template, and I wish to add only changes required for multiaz.

As it is a template, you always can change it afterwards (or use helm charts).

runc installation (& everything that is linked to the OMI) will be scoped in a future release.

Reply:
As a side note, you'll soon see that subnet declaration will be replaced by a single subregions declaration, saving many more lines...